### PR TITLE
Implement changesets on database level

### PR DESCRIPTION
**Pull request by ikalnytskyi**
Changesets are upcoming feature of XSnippet that allows to track changes
to snippets made by their owners. In order to avoid number of database
migrations, let's move onto new scheme right now and introduce basic
support.

The code is ugly now, yes, but I promise to refactor it in near future!

### CODE REVIEW

File: xsnippet/api/services/snippet.py [hunk 1]
```diff
@@ -47,23 +47,41 @@ def __init__(self, db):
...
         snippet['id'] = snippet_id
+        snippet['content'] = snippet.pop('changesets', [])[0]['content']
+
         return snippet
 
     async def update(self, snippet):
         now = datetime.datetime.utcnow().replace(microsecond=0)
         snippet['updated_at'] = now
 
+        condition = {'$set': snippet}
```
Comment (malor) [inline xsnippet/api/services/snippet.py]: nit: it's not a condition, it's the `update` parameter - a document that describes what updates must be performed

File: xsnippet/api/services/snippet.py [hunk 2]
```diff
@@ -111,7 +129,11 @@ def __init__(self, db):
             ('created_at', sort['created_at']),
             ('_id', sort['_id']),
         ])
-        return await query.limit(limit).to_list(None)
+
+        snippets = await query.limit(limit).to_list(None)
+        for snippet in snippets:
+            snippet['content'] = snippet.pop('changesets', [])[-1]['content']
```
Comment (malor) [inline xsnippet/api/services/snippet.py]: nit: maybe we should leave this for the API layer, but we can discuss this later, when we actually start using the changesets

File: tests/resources/test_snippets.py [hunk 3]
```diff
@@ -18,13 +18,15 @@
 
 
 @pytest.fixture(scope='function')
-def snippets():
+async def snippets(db):
```
[the remark this answers is not in the document]
Comment (ikalnytskyi) [reply tests/resources/test_snippets.py]: It happens due to `pytest-aiohttp` that provides necessary hooks to run this :)

File: xsnippet/api/services/snippet.py (hunk 1 above)
Comment (ikalnytskyi) [reply xsnippet/api/services/snippet.py]: Will rename

File: xsnippet/api/services/snippet.py (hunk 2 above)
Comment (ikalnytskyi) [reply xsnippet/api/services/snippet.py]: I was thinking about it and agree it probably would be better this way.. I don't remember why I decided to put it here for now, but I bet there were a reason. I think we will change this place plenty of times while we really start using it :)

**Pull request merged**